tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -155,6 +165,14 @@ public async Task Sanitize(RecordedTestSanitizer sanitizer, bool shouldLock = tr
                 }
             }
         }
+        private void LogSanitizerModification(string sanitizerId, string entryPreSanitize, string entryPostSanitize)
+        {
+            var result = StringComparer.OrdinalIgnoreCase.Compare(entryPreSanitize, entryPostSanitize);
```

Comment:
perhaps we could detect that the entry changed without a string comparison? I assume .NET has some pattern for flagging when objects are modified?

Reply:
yeah, I did think that. It's just my C# inexperience showing here :P

Will update soon/

Reply:
Updated to track the property changes in the setters.
Let me know if you like that better.

Reply:
it's hard to track the object changes through setters in cases where the value for a certain key is modified in the dictionary.
Sample snippet..
https://www.programiz.com/online-compiler/6lO13l7zzXYZU
 
Essentially, 
        `obj2.Headers["key2"] = obj2.Headers["key2"].Select(x => x.Replace(x, x+"_modified")).ToArray();`
 
In this case, Headers 's setter is not invoked, which means my modify-flag will not change.
 
The `RequestUri` and the `Body` are not affected the same way.
Still, for consistency, removed the modify-flag tracking in the setters in favor of doing the comparisons in Debug mode.